### PR TITLE
[BEAM-9199] Require Dataflow --region in Python SDK.

### PR DESCRIPTION
**Pull request by ibzib**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by ibzib** [comment]
Run Portable_Python PreCommit

**Comment by ibzib** [comment]
Run Python PreCommit

File: sdks/python/apache_beam/options/pipeline_options_validator_test.py [hunk 1]
```diff
@@ -38,9 +38,12 @@
 # Mock runners to use for validations.
 class MockRunners(object):
   class DataflowRunner(object):
-    pass
+    def get_default_gcp_region(self):
```
Comment (aaltay) [inline sdks/python/apache_beam/options/pipeline_options_validator_test.py:41]: Clever, but a bit dangreous.

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 2]
```diff
@@ -1396,6 +1399,36 @@ def json_string_to_byte_array(encoded_string):
     """Implements org.apache.beam.sdk.util.StringUtils.jsonStringToByteArray."""
     return unquote_to_bytes(encoded_string)
 
+  def get_default_gcp_region(self):
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py:1402]: Do we test this method? (Tests all seemingly overwrite it.)

File: sdks/python/apache_beam/options/pipeline_options_validator_test.py (hunk 1 above)
Comment (ibzib) [reply sdks/python/apache_beam/options/pipeline_options_validator_test.py:41]: We could also just add `--region` to all the tests if you would prefer.

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py (hunk 2 above)
Comment (ibzib) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py:1402]: We test the Java equivalent, but not Python. I suppose we should.

File: sdks/python/apache_beam/options/pipeline_options_validator_test.py (hunk 1 above)
Comment (aaltay) [reply sdks/python/apache_beam/options/pipeline_options_validator_test.py:41]: Not really. I like your solution better.

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py (hunk 2 above)
Comment (aaltay) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py:1402]: I think we should, if we can.

Comment (ibzib) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py:1402]: Added some unit tests, PTAL.

**Comment by ibzib** [comment]
I found quite a few more examples where --region needed to be added. Because this is a breaking change and it's very likely I've missed places, I will send an email to the dev list so that everyone is aware of this change (and the upcoming changes to Java and Go).

**Comment by aaltay** [comment]
Still LGTM. Thanks for the other updates.

**Comment by ibzib** [comment]
Failure in `hdfsIntegrationTest` looks like known flake (BEAM-7405 et al): `docker-credential-gcloud not installed or not available in PATH`

**Pull request merged**